src/backend/src/app.js

```diff
@@ -47,27 +57,46 @@ process.on('exit', (code) => {
  * @param {Buffer} options.rootKey
  * @param {string} options.migrationsFolderPath
  * @param {string} options.sharedStoragePath Path to app-specific external file storage folder
+ * @param {string} options.fallbackMapPath Path to app-specific external file storage folder
  *
  */
 export async function init({
   version,
   rootKey,
   migrationsFolderPath,
   sharedStoragePath,
+  fallbackMapPath,
 }) {
   log('Starting app...')
   log(`Device version is ${version}`)
 
   const privateStorageDir = rnBridge.app.datadir()
   const dbDir = join(privateStorageDir, DB_DIR_NAME)
   const indexDir = join(privateStorageDir, CORE_STORAGE_DIR_NAME)
+  const staticStylesDir = join(sharedStoragePath, 'styles')
 
   mkdirSync(dbDir, { recursive: true })
   mkdirSync(indexDir, { recursive: true })
+  mkdirSync(staticStylesDir, { recursive: true })
 
   const fastify = Fastify()
   const fastifyController = new FastifyController({ fastify })
 
+  // Register maps plugins
+  fastify.register(MapeoStaticMapsFastifyPlugin, {
+    prefix: 'static',
+    staticRootDir: staticStylesDir,
+  })
+  fastify.register(MapeoOfflineFallbackMapFastifyPlugin, {
+    prefix: 'fallback',
+    styleJsonPath: join(fallbackMapPath, 'style.json'),
+    sourcesDir: join(fallbackMapPath, 'dist'),
+  })
+  fastify.register(MapeoMapsFastifyPlugin, {
+    prefix: 'maps',
+    defaultOnlineStyleUrl: DEFAULT_ONLINE_MAP_STYLE_URL,
+  })
+
```

Comment:
it's to add maps functionality to the fastify instance that we use for serving things over http. the functionality is separate into 3 plugins:

- static: provides ability to statically serve custom offline maps
- fallback: provides ability to statically serve a directory that acts as the fallback map (which are set up differently that custom offline maps). this is mostly to accommodate for the fallback map structure that's used by Mapeo Mobile.
- maps: currently acts as the main resolver for the user-facing API e.g. resolving the appropriate style.json based on existence of static/fallback maps. in the future, this will probably hold other functionality related to features that are more similar to the Map Manager in Mapeo Mobile.